STMicroelectronics/STM32L4_L4+/app/az_ulib_dm/inc/az_ulib_registry_api.h

```diff
@@ -0,0 +1,173 @@
+ * adding a new key value pair to the registry.
+ *
+ * @param[in]   key                  The #az_span key to add to the registry.
+ * @param[in]   value                The #az_span value to add to the registry.
+ *
+ * @pre         \p key                            shall not be `#AZ_SPAN_EMPTY`.
+ * @pre         \p sizeof(key)                    shall greater than zero.
+ * @pre         \p value                          shall not be `#AZ_SPAN_EMPTY`.
+ * @pre         \p sizeof(value)                  shall greater than zero.
+ *
+ * @return The #az_result with the result of the registry operations.
+ *      @retval #AZ_OK                            If adding a value to the registry was successful
+ *      @retval #AZ_ERROR_ULIB_ELEMENT_DUPLICATE  If there is a key within the registry that is the
+ *                                                same as the new key.
+ *      @retval #AZ_ERROR_ULIB_SYSTEM             If the `az_ulib_registry_add` operation failed on
+ *                                                the system level.
+ *      @retval #AZ_ERROR_ULIB_BUSY               If the resources necesary for the
+ *                                                `az_ulib_registry_add` operation are busy.
+ *      @retval #AZ_ERROR_OUT_OF_MEMORY           If the flash space for `az_ulib_registry_add`
+ *                                                is not enough for a new registry entry.
+ */
+AZ_NODISCARD az_result az_ulib_registry_add(az_span key, az_span value);
+
+/**
+ * @brief   This function removes an #az_span key and it's corresponding #az_span value from the
+ * device registry.
+ *
+ * This function goes through the registry and marks the node with the given key value pair as
+ * deleted so that it can no longer be accessible.
+ *
+ * @param[in]   key                  The #az_span key to remove from the registry.
+ *
+ * @pre         \p key                            shall not be `#AZ_SPAN_EMPTY`.
+ * @pre         \p sizeof(key)                    shall greater than zero.
+ *
+ * @return The #az_result with the result of the registry operations.
+ *      @retval #AZ_OK                            If removing a value to the registry was successful
+ *      @retval #AZ_ERROR_ULIB_SYSTEM             If the `az_ulib_registry_delete` operation failed
+ *                                                on the system level.
+ *      @retval #AZ_ERROR_ULIB_BUSY               If the resources necesary for the
+ *                                                `az_ulib_registry_delete` operation are busy.
+ */
+AZ_NODISCARD az_result az_ulib_registry_delete(az_span key);
+
+/**
+ * @brief   This function initializes the device registry.
+ *
+ * This function initializes components that the registry needs upon reboot.
+ *
+ * @return The #az_result with the result of the registry operations.
+ *      @retval #AZ_OK                            If initializing the registry was successful.
+ */
+AZ_NODISCARD az_result az_ulib_registry_init();
```

Comment:
>az_result [](http://example.com/codeflow?start=13&length=9)

this functino doesn't look like it has the possibility to fail. consider making it void.

Reply:
Good call, will do. Thanks!